### PR TITLE
fix docker push github action to pass build context

### PR DESCRIPTION
**Pull request by darren-reddick**
Pull request to fix the github actions docker push action which was failing as the docker build context was not set properly.

Also some minor housekeeping on the github actions definition file.

### CODE REVIEW

**Comment by darren-reddick** [comment]
Coveralls step seems to broken despite no change to the command there??

**Comment by Ottovsky** [comment]
I am unsure why it is happening, the change itself looks ok.

**Comment by Ottovsky** [comment]
Actually, comparing to other repos, I don't think coveralls is working on the external PRs, maybe it is worth to add condition to this step to not run it on the pull requests.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -33,39 +33,29 @@ jobs:
...
     #
     # Component test
-    - run: ./test/test.sh component.py
+    - name: Run component tests
+      run: ./test/test.sh component.py
 
     #
     #
     # Upload coverage results to coveralls
     - name: Submit Coveralls
```
Comment (Ottovsky) [inline .github/workflows/main.yml:54]: Please add condition to not execute this step on external PRs.

**Review by Ottovsky — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (darren-reddick) [reply .github/workflows/main.yml:54]: Done

**Review by Ottovsky — APPROVED** [review]
lgtm

**Pull request merged**